src/options.js

```diff
@@ -80,6 +80,16 @@ export function arrayify(data, type) {
     : (data instanceof type ? data : type.from(data)));
 }
 
+// An optimization of type.from(values, f): if the given values are already an
+// instanceof the desired array type, the faster values.map method is used.
+export function map(values, f, type = Array) {
+  return values instanceof type ? values.map(f) : type.from(values, f);
```

Comment:
If `values` is indexable and has a length, could the same for-loop approach as in [this PR](https://github.com/observablehq/plot/pull/861/files#diff-b3c78ec4a30606d2f642a7425509de6e2942d4ff24f8304d3f40f2aadacad887R130) speed up the "else" branch?

Then we could avoid the slower iterator-based `type.from` unless we find ourselves in the general iterator case.

Reply:
> If values is indexable and has a length

There’s not really a way to test whether a value is “indexable”, and checking for a length property is not sufficient. For example, if `values` is a string, it has a length property and is indexable, but you wouldn’t want to use that here. We could do an `instanceof Array` test and use the faster form, but that would only help in the case of numeric coercion, and in that case, it would still be faster for the caller to provide a typed array and avoid the coercion entirely.

Reply:
That’s a really good and clarifying example. Thanks for explaining!